python/Ganga/new_tests/GPI/Loading/LazyLoadingTest.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+class LazyLoadingTest(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(LazyLoadingTest, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs
+        j=Job()
+        self.assertTrue(len(jobs) == 1)
+
+    def test_b_JobNotLoaded(self):
+        """ Second get the job and check that getting it via jobs doesn't cause it to be loaded"""
+        from Ganga.GPI import jobs
+
+        self.assertTrue(len(jobs) == 1)
+
+        print("len: %s" % str(len(jobs)))
+
+        j = jobs(0)
+
+        from Ganga.GPIDev.Base.Proxy import stripProxy
+        raw_j = stripProxy(j)
+
+        has_loaded_job = raw_j._getRegistry().has_loaded(raw_j)
+
+        self.assertTrue(has_loaded_job == False)
```

Comment:
It's a first draft to see if it worked, I'll spruce it up tomorrow at some point